leftwm/src/config.rs

```diff
@@ -97,13 +97,17 @@ impl WindowHook {
     /// Multiple [`WindowHook`]s might match a `WM_CLASS` but we want the most
     /// specific one to apply: matches by title are scored greater than by `WM_CLASS`.
     fn score_window(&self, window: &Window) -> u8 {
-        let class_score = match (&self.window_class, &window.res_class) {
-            (Some(wc_re), Some(res_class)) => u8::from(wc_re.replace(res_class, "") == ""),
+        let class_score = match (&self.window_class, &window.res_class, &window.res_name) {
+            (Some(wc_re), Some(res_class), Some(res_name)) => {
```

Comment:
Without me knowing any better; Can we assume that `window.res_class` and `window.res_name` are always both present? Because if one of them doesn't, this match arm is not reached. This might potentially fix the regression but introduce another.

Reply:
Your updated code should work, the only downside is that there is some duplicate code and that the syntax is a bit hard to understand at a glance. But the syntax was hard to understand to begin with IMO. I will give you a suggestion on how I'd refactor the method, but there really is no right or wrong and it mostly comes down to personal preference.